gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -84,11 +84,20 @@
  */
 @InternalExtensionOnly
 public final class InstantiatingGrpcChannelProvider implements TransportChannelProvider {
```

Comment:
This is a final class and I'm unable to subclass this in the unit tests. For now, I've decided to keep the final keyword and workaround this by including package-private setters for the EnvProvider and the SystemProductNameReader.

Reply:
Did you mean that I should remove the annotation? Or that I should keep the everything the same?

Reply:
We should remove the annotation.